build.gradle

```diff
@@ -195,15 +195,13 @@ task getRevision() {
 	onlyIf {
 		file('.git').exists()
 	}
+	def commit = System.getProperty("commit") ?: 'origin/main'
```

Comment:
will this fail if origin is not a remote repository for this clone?

Reply:
It was already using origin/main I didn't change that. If you think it's an issue please address it in a different pr.

Reply:
Set either the ORG_GRADLE_PROJECT_commit environment variable to a commit hash, or pass -Pcommit=HASH on the command line

Reply:
To add a little more context:

We need some way of determining what the canonical branch is. We can't just use the tracked branch -- otherwise, say, this feature branch with 6 non-mainline commits would be r+6.

I experienced issues with this when @Rinn initially pushed the change to use git rev-list, but it was a simple enough change to rename the corresponding remote.

Reply:
yeah idk i'd need to think about it more, maybe it can just check if the current commit is tagged with a revision and if it's not consider it dirty.